api_views/users.py

```diff
@@ -206,3 +206,64 @@ def delete_user(username):
                 return Response(error_message_helper("User not found!"), 404, mimetype="application/json")
         else:
             return Response(error_message_helper("Only Admins may delete users!"), 401, mimetype="application/json")
+
+
+def get_user_by_email(email):
+    """
+    Retrieve user details by email address.
+    """
+    user = User.query.filter_by(email=email).first()
+    if user:
+        return jsonify(user.json()), 200
+    else:
+        return Response(error_message_helper(), 404, mimetype="application/json")
+
+
+def promote_user_to_admin(username):
```

Comment:
Missing unit tests for the promote_user_to_admin function. Add tests to verify admin promotion functionality.